torch/_export/serde/serialize.py

```diff
@@ -587,13 +586,22 @@ def export_nn_module_stack(val):
                 assert isinstance(path, str)
                 assert isinstance(ty, str)
 
-                return path + "," + ty
+                return (path, ty)
 
-            # Serialize to "key,orig_path,type_str"
-            nn_module_list = [
-                f"{k},{export_nn_module_stack(v)}" for k, v in nn_module_stack.items()
-            ]
-            ret["nn_module_stack"] = ST_DELIMITER.join(nn_module_list)
+            keys, paths, tys = [], [], []
+            for k, v in nn_module_stack.items():
+                keys.append(k)
+                path, ty = export_nn_module_stack(v)
+                paths.append(path)
+                tys.append(ty)
+
+            nn_module_stack_dict = {
+                "key_list": keys,
+                "path_list": paths,
+                "type_list": tys
+            }
```

Comment:
I was actually thinking the schematized nn_module_stack would look something like:

```
@dataclass
class NNModuleInfo:
    key: str
    path: str
    type: str

serialized_nn_module_stack = json.dump([NNModuleInfo(key, path, type), ...])
```

Since nn_module_stack is an ordered dict, this would also preserve the ordering later on.

Reply:
that was actually my first attempt but then I realized that a `dataclass` can not be serialized with `json.dumps()` unless it is converted to a dictionary using `asdict()`. So is adding this `dataclass` in the `schema.py` kind of extra?

Reply:
ok, that makes sense. This is fine then!